fuels-abigen-macro/tests/harness.rs

```diff
@@ -1075,6 +1082,27 @@ async fn test_large_return_data() {
     // Two word-sized string
     let res = contract_instance.get_large_string().call().await.unwrap();
     assert_eq!(res.value, "ggggggggg");
+
+    // Large struct will be bigger than a `WORD`.
+    let res = contract_instance.get_large_struct().call().await.unwrap();
+    assert_eq!(res.value.foo, 12);
+    assert_eq!(res.value.bar, 42);
+
+    // Array will be returned in `ReturnData`.
+    let res = contract_instance.get_large_array().call().await.unwrap();
+    assert_eq!(res.value, &[1, 2]);
+
+    let res = contract_instance.get_contract_id().call().await.unwrap();
+
+    // First `value` is from `CallResponse`.
+    // Second `value` is from Sway `ContractId` type.
+    assert_eq!(
+        res.value.value,
+        [
+            255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+            255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255
+        ]
+    );
```

Comment:
@nfurfaro I believe this is the relevant case for you: `get_contract_id()` returns the Sway type `ContractId`, which wraps the `b256` in a `Struct`, which wasn't being handled before. Now it seems to be working fine!